### ISSUE TITLE
[aws_cognito_identity_pool_roles_attachment mapping_rule] validation if type=Token

### ISSUE THREAD

**Issue by mnwk** [terraform-providers/terraform-provider-aws#2232]
```
role_mapping {
    identity_provider = “providername”
    ambiguous_role_resolution = “AuthenticatedRole”
    type = “Token”
  }
```
leads to:
```
aws_cognito_identity_pool_roles_attachment.main: Error validating ambiguous role resolution: [Role Mapping “5u5pf3fsqvd45ec4agl2snsf0l”: [mapping_rule must not be set for Token based role mapping]]
```
"[mapping_rule must not be set..." and it isn't set, so there should be no problem.. but it is.

maybe forgot "if ( **!** ok || valLength > 0)" here:

https://github.com/terraform-providers/terraform-provider-aws/blob/e398f974ac187ce580494f27efff28a25d8585b0/aws/validators.go#L1585-L1587;

**Comment by bflad** [comment]
The fix for this has been merged into master and will be released in v1.9.0 of the AWS provider, expected at the end of the week. 🎉

**Comment by bflad** [comment]
This has been released in [terraform-provider-aws version 1.9.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#190-february-09-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
